File: Selenium/Project/CRM_Activity8.java
package project;

import org.testng.annotations.Test;
import org.testng.annotations.BeforeClass;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Reporter;
import org.testng.annotations.AfterClass;

public class CRM_Activity8 {
	
	WebDriver driver;
	Actions builder;
	WebDriverWait wait;
	
  @Test
  public void traverseTable() {
	  
	//Login
		 CRM_Activity4 login = new CRM_Activity4();
		 login.verifyLogin(driver);
	  
	  //Navigate to Sales
	  
	  wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("grouptab_0")));
	  WebElement sales = driver.findElement(By.id("grouptab_0"));
	  builder.moveToElement(sales).build().perform();
	  
	  //Select Accounts option
	  driver.findElement(By.xpath("*//a[@id='grouptab_0']/following-sibling::ul/li[2]")).click();
	  
	  wait.until(ExpectedConditions.visibilityOfAllElementsLocatedBy(By.xpath("*//table[contains(@class,'table-responsive')]")));
	  
	  //Printing names from table
	  for(int i=1; i<=10 ; i++)
	  {
		  if (i%2 == 1)
				  {
	  String nameText = driver.findElement(By.xpath("*//table[contains(@class,'table-responsive')]/tbody/tr["+i+"]/td[3]")).getText();
	  System.out.println(nameText);
				  }
	  }
  }
  @BeforeClass
  public void beforeClass() {
	  
	  driver = new ChromeDriver();
	  builder = new Actions(driver);
	  wait = new WebDriverWait(driver,Duration.ofSeconds(10));
  }

  @AfterClass
  public void afterClass() {
	  
	  driver.quit();
  }

}

File: Selenium/Project/CRM_Activity8.py
# Import 
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from CRM_Activity4 import verify_login
from selenium.webdriver import ActionChains

# Start the Driver
with webdriver.Chrome() as driver:
    # Navigate to the URL
        driver.get("https://alchemy.hguy.co/crm/")

        #Login
        verify_login(driver)

        #Navigate to sales
        wait = WebDriverWait(driver,10)
        wait.until(EC.visibility_of_element_located((By.ID,"grouptab_0")))
        sales = driver.find_element(By.XPATH,"*//a[@id='grouptab_0']")
        actions = ActionChains(driver)
        actions.move_to_element(sales).click().perform()

        #Select accounts option
        driver.find_element(By.XPATH,"*//a[@id='grouptab_0']/following-sibling::ul/li[2]").click()
        wait.until(EC.visibility_of_element_located((By.XPATH,"*//table[contains(@class,'table-responsive')]")))

        #Printing odd rows names from table
        for i in range(1, 11):
                if i % 2 == 1: 
                        name_text = driver.find_element(By.XPATH, f"//table[contains(@class,'table-responsive')]/tbody/tr[{i}]/td[3]").text
                        print(name_text)

driver.quit()